src/pages/journal/components/empty-slot/empty-slot.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, Input } from '@angular/core';
+import { SlotComponent } from '../slot/slot';
+import { ScreenOrientation } from '@ionic-native/screen-orientation';
+
+@Component({
+  selector: 'empty-slot',
+  templateUrl: 'empty-slot.html',
+})
+export class EmptySlotComponent implements SlotComponent {
+  @Input()
+  slot: any;
+
+  @Input()
+  hasSlotChanged: boolean;
+
+  @Input()
+  showLocation: boolean;
+
+  constructor(public screenOrientation: ScreenOrientation) {}
+
+  isPortrait() : boolean {
```

Comment:
Can we move this up to SlotComponent and then remove it from TestSlotComponent

Reply:
SlotComponent is an interface not a class so can't move the method into the interface.
(I know in C# 7.0 (or later) they are allowing this, but did a quick check on typescript and this isn't supported.

Reply:
so many languages with so many different rules . Happy to close this one